File: system-test/typescript/src/testkit/driver/adapter/external/tax/tax-real-driver.ts
import type { Result } from '../../../../common/result.js';
import { success } from '../../../../common/result.js';
import type { TaxErrorResponse } from '../../../port/external/tax/dtos/TaxErrorResponse.js';
import type { ReturnsTaxRateRequest } from '../../../port/external/tax/dtos/ReturnsTaxRateRequest.js';
import { BaseTaxDriver } from './BaseTaxDriver.js';
import { TaxRealClient } from './client/TaxRealClient.js';

export class TaxRealDriver extends BaseTaxDriver<TaxRealClient> {
  constructor(baseUrl: string) {
    super(new TaxRealClient(baseUrl));
  }

  async returnsTaxRate(_request: ReturnsTaxRateRequest): Promise<Result<void, TaxErrorResponse>> {
    return success(undefined);
  }

  async close(): Promise<void> {}
}

File: system-test/java/src/main/java/com/optivem/shop/testkit/driver/adapter/external/tax/TaxRealDriver.java
package com.optivem.shop.testkit.driver.adapter.external.tax;

import com.optivem.shop.testkit.driver.adapter.external.tax.client.TaxRealClient;
import com.optivem.shop.testkit.driver.port.external.tax.dtos.ReturnsTaxRateRequest;
import com.optivem.shop.testkit.driver.port.external.tax.dtos.error.TaxErrorResponse;
import com.optivem.shop.testkit.common.Result;

public class TaxRealDriver extends BaseTaxDriver<TaxRealClient> {
    public TaxRealDriver(String baseUrl) {
        super(new TaxRealClient(baseUrl));
    }

    @Override
    public Result<Void, TaxErrorResponse> returnsTaxRate(ReturnsTaxRateRequest request) {
        return Result.success();
    }
}
